frontend/src/components/authchooser/index.tsx

```diff
@@ -81,7 +81,9 @@ function AuthChooser(){
 
     // If we haven't yet figured whether we need to use a token for the current
     // cluster, then we check here.
-    if (cluster.useToken === undefined) {
+    // oidc should first see if they want to try and login.
+    const noLoginCheck = (clusterAuthType === 'oidc' && testingAuth || clusterAuthType !== 'oidc');
```

Comment:
Shouldn't we simply check for `clusterAuthType !== 'oidc'`?
Or (probably better) just not redirect to the token prompt within the next if block when the clusterAuthType is `oidc`?

The overall problem seems to be that this logic was for checking whether we use a certificate (i.e. no token required) or need a token, but the oidc option has been forgotten here. Redirecting to the token directly is what we need iff the token is the only auth option (i.e. we don't want to show a dialog with just one option to the user).

Reply:
Yes, you're right. I changed it to just do: `if (clusterAuthType !== 'oidc' && cluster.useToken === undefined)`.

Reply:
Both clicking the signin button, and also entering a token worked with this change.